File: backend/src/routers/architecture.py
from fastapi import APIRouter

from models.schemas import ArchitecturePlanRequest
from services.architecture_planner import create_architecture_plan

router = APIRouter(prefix="/api/architecture", tags=["architecture"])


@router.post("/plan")
def plan_architecture(request: ArchitecturePlanRequest) -> dict[str, object]:
    return create_architecture_plan(request)

File: frontend/src/api/architecture.ts
import {api} from './client'
import type {ArchitecturePlan,DeploymentResourceKey,OperationalRequirements} from './models'

export const planArchitecture=(request:{resources:DeploymentResourceKey[];primaryRegion?:string;operationalRequirements:OperationalRequirements})=>api<ArchitecturePlan>('/api/architecture/plan',{method:'POST',body:JSON.stringify(request)})